patches/build-config-BUILDCONFIG.gn.patch

```diff
@@ -10,3 +10,9 @@ index 034999175e30316e488404aa5392f38792d8bdde..e10c8b859e46bc802e24ce83e830e86b
    "//build/config:feature_flags",
    "//build/config/compiler:afdo",
    "//build/config/compiler:afdo_optimize_size",
+@@ -594,3 +595,5 @@ set_defaults("component") {
+     configs = default_compiler_configs
+   }
+ }
++
++import("//brave/build/config/brave_build.gni")
```

Comment:
why is this change needed?

Reply:
It's not a strictly necessary change for this task, but it improves the idea of our automatic `gni` includes to reduce patching.
For a moment I was thinking to do this in a separate PR/issue, but decided to keep it's here for now. I can change this if someone also thinks it's not necessary here.

Reply:
Had to remove this, because included variables are visible everywhere and it breaks our gn-rust integration.